### PR TITLE
SSO Changes

### PR DESCRIPTION
**Pull request by himalay-goswami**
Made the following changes:

1. Added a new API in TL-Controller.
2. Added SSO_ENDPOINT filtering in AuthPreCheckFilter and RbacPreCheckFilter.
3. Added custom auth filter SsoAuthFilter (and model entities associated with the filter).

### CODE REVIEW

File: core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+    private final User user;
+
+    private final ObjectMapper objectMapper;
+
+    private final RestTemplate restTemplate;
+
+    public static final String SSO_ENDPOINT = "/tl-services/external/_create";
+
+    public static final String THIRD_PARTY_URL = "https://eservicesuk-staging.prodios.com/api/token/verify/";
```
Comment (vinothrallapalli-eGov) [inline core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java]: Move this into application.property file and read it from there.

File: core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+        //Step 1: Get header from 3rd party API
+        String header = getJwtToken(context);
+
+        //step 2: verify token and get JSON response
+        SsoUserEntity responsePojo = verifyToken(header, context);
+
+        //step 2.1 Get user info from responsePojo, and then phoneNumber
+        UserInfo userInfo = responsePojo.getUser();
+        String phoneNumber = userInfo.getPhoneNumber();
```
Comment (vinothrallapalli-eGov) [inline core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java]: Rename variable phoneNumber to mobileNumber.

File: frontend/mono-ui/web/rainmaker/yarn.lock [hunk 3]
```diff
@@ -4,49 +4,49 @@
 
 "@babel/code-frame@^7.0.0":
   version "7.10.4"
-  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.10.4.tgz#168da1a36e90da68ae8d49c0f1b48c7c6249213a"
+  resolved "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.10.4.tgz"
```
Comment (vinothrallapalli-eGov) [inline frontend/mono-ui/web/rainmaker/yarn.lock]: Why you have changed from yarn to npm. Please revert it.

File: core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java (hunk 1 above)
Comment (himalay-goswami) [reply core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java]: These changes are done.

File: core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java (hunk 2 above)
Comment (himalay-goswami) [reply core-services/zuul/src/main/java/org/egov/filters/pre/SsoAuthFilter.java]: This is changed to getMobileNumber()

File: frontend/mono-ui/web/rainmaker/yarn.lock (hunk 3 above)
Comment (himalay-goswami) [reply frontend/mono-ui/web/rainmaker/yarn.lock]: I do not remember changing this but I have reverted to yarn.

**Pull request closed without merging**